pkg/service/restore/index.go

```diff
@@ -0,0 +1,194 @@
+// Copyright (C) 2024 ScyllaDB
+
+package restore
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+	. "github.com/scylladb/scylla-manager/v3/pkg/service/backup/backupspec"
+	"github.com/scylladb/scylla-manager/v3/pkg/sstable"
+)
+
+// LocationWorkload represents aggregated restore workload
+// in given backup location.
+type LocationWorkload struct {
+	Location
+
+	Size   int64
+	Tables []TableWorkload
+}
+
+// TableWorkload represents restore workload
+// from many manifests for given table in given backup location.
+type TableWorkload struct {
+	Location
+	TableName
+
+	Size       int64
+	RemoteDirs []RemoteDirWorkload
+}
+
+// RemoteDirWorkload represents restore workload
+// for given table and manifest in given backup location.
+type RemoteDirWorkload struct {
+	TableName
+	*ManifestInfo
+
+	RemoteSSTableDir string
+	Size             int64
+	SSTables         []RemoteSSTable
+}
+
+// RemoteSSTable represents SSTable updated with size and version info from remote.
+type RemoteSSTable struct {
+	SSTable   // File names might contain versioned snapshot tag extension
+	Size      int64
+	Versioned bool
+}
+
+// SSTable represents files creating a single sstable.
+type SSTable struct {
+	ID    string
+	Files []string
+}
+
+// IndexWorkload returns sstables to be restored aggregated by location, table and remote sstable dir.
+func (w *tablesWorker) IndexWorkload(ctx context.Context, locations []Location) ([]LocationWorkload, error) {
```

Comment:
Could it be logged somehow ? I mean the LocationWorkload without including name and size of every SSTable, but just size of all SSTables in the current chunk + amount of SSTables.
Would it help somehow with debugging ?